### ISSUE TITLE
Error handling for dukcs/fetch

### ISSUE THREAD

**Issue by skovhus** [Unity-Technologies/Tanto#27]
Seems we have an issue with error handling for the new `ducks/fetch` + `fetchAnythingSaga`.

Example http://localhost:3000/project/unity/pullrequest/1 on `fixes-due-api-changes`. The graphql error is not propagated correctly and the code is rather hard to debug. The graphql error is  that `line` should be `lineNumber` in the `universal/services/ono/queries/pullRequest/index.js`

![screen shot 2017-01-16 at 12 07 48](https://cloud.githubusercontent.com/assets/1260305/21980663/716090ae-dbe4-11e6-8013-f70855b40794.png)

Investigation:
- Problem is that the `SET_REPOSITORY` reducer inside `ducks/repositories` is running although the request failed...
- A bit deeper: the `fetchActionCreator` (`ducks/fetch`) callback is called although the request failed.

@kkateq Did the changes for `fetchAnythingSaga` + `ducks/fetch` ignore error case handling or am I missing something? We used to handle this.

**Comment by kkateq** [comment]
It's not about failed request, it unstable and now depends on PR query response time. There some section in metadata that tries to render PR metadata info from store that still in fetching progress, if fetch is fast enough - it renders perfectly ;)  I am working on this issue by splitting pull request query on few several ones and make each different parts of PR layout to be queried separately(as well as files list) as part of querying files file by file. So that user shouldn't wait until the whole PR fetched to see metadata for example.

**Comment by skovhus** [comment]
Hm. Then it is also about failing request. The pull request query fails, but no error is propagated to the UI (i.e. `error` is empty, hence we show the 404 screen on error).

![screen shot 2017-01-16 at 12 21 50](https://cloud.githubusercontent.com/assets/1260305/21981151/64d8fe28-dbe6-11e6-8fdc-ee1c473c5c5f.png)

**Comment by kkateq** [comment]
@skovhus Corrected above. Might be another issue.

**Comment by skovhus** [comment]
Can you describe how you imagined error handling working for `fetchActionCreator`? We currently  just pass in one callback... Either the callback should take two arguments (node style with data and error) or we need another mechanisms for errors.

**Comment by kkateq** [comment]
Nothing to do with `SET_REPOSITORY` here it was called before when you browsed the project. When you open PR `SET_PULL_REQUEST` should be called, if the request is successful, as I can see it's not.

`fetchActionCreator` wraps action and handle any query `isFetching` progress and `error` separately by `fetch` reducer where you can find a map and get error result by action name. That gives us the advantage to track fetch status and errors separately for any PullRequest partial data query outside `pullrequests` reducer (where otherwise we would have only one field for `error` and `isFetching` which would block partial rendering of PR layout until all the pull request queries finish). Check out `ducks/fetch` reducer.

**Comment by kkateq** [comment]
And in the case of exception inside callback, it should be caught inside fetch saga and appear and be processed by `fetch` reducer.

**Comment by skovhus** [comment]
I've tracked the bug down.

https://github.com/Unity-Technologies/Tanto/blob/master/src/client/ducks/fetch/index.js#L31 

`SENDING_REQUEST` wrongly replaces the `error` object.

**Comment by skovhus** [comment]
Same problem with `REQUEST_ERROR` and `CLEAR_ERROR`. I'll fix and add a test.
